arrow-core/src/main/java/com/raphtory/arrowcore/implementation/VertexHistoryPartition.java

```diff
@@ -610,48 +605,43 @@ protected void findHistory(VertexHistoryIterator.WindowedVertexHistoryIterator s
     }
 
 
+    /**
+     * @return the lowest history time in this partition
+     */
     public long getLowestTime() {
         if (_history._maxRow==0) {
             return Long.MAX_VALUE;
         }
 
-        if (_historyRO.getRowCount() != _history._maxRow) {
-            _historyRO.setRowCount(_history._maxRow);
-        }
-
-        if (!_sorted) {
-            sortHistoryTimes();
-        }
-
+        sortHistoryTimes();
 
         int lowestRow = _history._sortedTimeIndices.get(0);
 
         return _history._times.get(lowestRow);
     }
 
 
+    /**
+     * @return the highest history time in this partition
+     */
     public long getHighestTime() {
         if (_history._maxRow==0) {
             return Long.MIN_VALUE;
         }
 
-        if (_historyRO.getRowCount() != _history._maxRow) {
-            _historyRO.setRowCount(_history._maxRow);
-        }
-
-        if (!_sorted) {
-            sortHistoryTimes();
-        }
+        sortHistoryTimes();
 
         int highestRow = _history._sortedTimeIndices.get(_history._maxRow-1);
 
         return _history._times.get(highestRow);
     }
 
 
+    /**
+     * @return the number of history items in this partition
+     */
     public long getNHistoryItems() {
-        int n = _history._maxRow;
-        return n+1;
+        return _history._maxRow;
     }
 
 
```

Comment:
//System.out.println("TIME at " + row + " : " + creationTime + ", inRange=" + (creationTime>=min && creationTime<=max)); to logger or removed?